Dockerfile

```diff
@@ -9,7 +9,7 @@ RUN apt update && \
 
 # Install pip and poetry
 RUN pip install --no-cache --upgrade pip
-RUN pip install --no-cache "poetry==1.3.2"
+RUN pip install --no-cache "poetry==1.6.1"
```

Comment:
I'm all for this (1.3.2 is like over a year old!), but this does makes this inconsistent with most of our other repos so might need attention as a separate bit of work, is it needed as part of this change?

Reply:
Also also! Any reason (if/when we do start changing these) not to go straight to the latest, just checked and there is a 1.8.2 available

Reply:
Hey, I was trying to build the Docker image but it was failing due to the Poetry version being out of step with the lockfile. I went with 1.6.1 as that's what is in CI for most repos and seems to be the version fixed in Dockerfiles in the other repos I'm regularly in - but I have no preference on whether we should be on this, 1.7.1 or higher/latest!

Reply:
The current precommits are checking for 1.7.1, so I'm going to bump this to 1.7.1 for now in this PR. We definitely need to standardise this across repos though - will likely add this as a custom precommit check in Trunk

Reply:
Sounds good, and looking forward to it!

Reply:
Just occured to me (in case you werent already planning to do this) that it might be worth double checking the image size after this change, its probably fine but we have some slightly hacky image size optmisations in the dockerfile so itd be worth ensuring they still work at later versions

Reply:
2.13GB :)

Reply:
Which is very similar to before, so no worries here! Fred and I have spoken in Slack about this in more detail and he is happy for me to resolve these conversations in the meantime